src/views/clusteroverview/OverviewTab/vms-per-resource-card/utils/utils.ts

```diff
@@ -0,0 +1,129 @@
+import shuffle from 'lodash.shuffle';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { instanceTypeSeriesNameMapper } from '@kubevirt-utils/components/AddBootableVolumeModal/components/VolumeMetadata/components/InstanceTypeDrilldownSelect/utils/constants';
+import { getInstanceTypePrefix } from '@kubevirt-utils/resources/bootableresources/helpers';
+import {
+  LABEL_USED_TEMPLATE_NAME,
+  LABEL_USED_TEMPLATE_NAMESPACE,
+} from '@kubevirt-utils/resources/template';
+import { LightMultiColorOrderedTheme } from '@patternfly/react-charts/dist/js/components/ChartTheme/themes/light/multi-color-ordered-theme';
+
+import { ChartDataObject } from './constants';
+import { RunningVMsChartLegendLabelItem } from '../RunningVMsChartLegendLabel';
+
+const swapArrayElems = (arr) => ([arr[0], arr[1]] = [arr[1], arr[0]]);
+
+export const getColorList = (numColors: number) => {
+  const originalColors = LightMultiColorOrderedTheme.pie.colorScale;
+  const numOriginalColors = originalColors.length;
+  const colorList = [].concat(originalColors);
+
+  if (numColors > numOriginalColors) {
+    // assemble an array of the required size by repeating colors
+    const fullMultiples = numColors / numOriginalColors;
+    const remainder = numColors % numOriginalColors;
+
+    // add full shuffled copies of the original list
+    for (let i = 1; i <= fullMultiples; i++) {
+      const shuffledColors = shuffle([].concat(originalColors));
+      if (shuffledColors[0] === colorList[colorList.length - 1]) {
+        // swap colors to avoid adjoining colors being equal
+        swapArrayElems(shuffledColors);
+      }
+      colorList.concat(shuffledColors);
+    }
+
+    // add a shuffled slice of the original list to meet length requirement
+    if (remainder) {
+      const shuffledRemainder = shuffle(originalColors.slice(0, remainder));
+      if (remainder > 1 && shuffledRemainder[0] === colorList[colorList.length - 1]) {
+        // swap colors to avoid adjoining colors being equal
+        swapArrayElems(shuffledRemainder);
+      }
+      colorList.concat(shuffledRemainder);
+    }
+  }
+
+  return colorList;
+};
+
+export const getInstanceTypeSeriesLabel = (instanceTypeName: string): string => {
+  const [seriesName] = instanceTypeName?.split('.');
```

Comment:
can this line be:
`const seriesName = getInstanceTypePrefix(instanceTypeName);`?

Reply:
It can't be because it will return me the first letter and I need the whole name to make the change :)